qcodes/instrument_drivers/Keysight/Keysight_E8267D.py

```diff
@@ -27,21 +28,39 @@ class Keysight_E8267D(VisaInstrument):
     def __init__(self, name, address, step_attenuator=False, **kwargs):
         super().__init__(name, address, **kwargs)
 
+        # Only listed most common spellings idealy want a
+        # .upper val for Enum or string
+        on_off_validator = vals.Enum('on', 'On', 'ON',
+                                     'off', 'Off', 'OFF')
         self.add_parameter(name='frequency',
                            label='Frequency',
                            unit='Hz',
                            get_cmd='FREQ:CW?',
                            set_cmd='FREQ:CW' + ' {:.4f}',
                            get_parser=float,
                            set_parser=float,
-                           vals=vals.Numbers(1e5, 20e9))
+                           vals=vals.Numbers(1e5, 20e9),
+                           docstring='Adjust the RF output frequency')
+        self.add_parameter(name='frequency_offset',
+                           label='Frequency offset',
+                           unit='Hz',
+                           get_cmd='FREQ:OFFS?',
+                           set_cmd='FREQ:OFFS {}',
+                           get_parser=float,
+                           vals=Numbers(min_value=-200e9,
+                                        max_value=200e9))
+        self.add_parameter('frequency_mode',
+                           label='Frequency mode',
+                           set_cmd='FREQ:MODE {}',
+                           get_cmd='FREQ:MODE?',
+                           vals=vals.Enum('FIX', 'CW', 'SWE', 'LIST'))
```

Comment:
same comment here as for the "IQsource#" parameters - would human-readable values be better here? say (just guessing) "fixed" for "FIX", "continuous_wave" for "CW", etc? and a nice `docstring=...`?

Reply:
I copied this one from the agilent E8267 driver. But it might make sense to change the names now.

Reply:
ok, it's fine. And thanks for the docstring.